### DISCUSSION TITLE
Help wrapping

### DISCUSSION THREAD

**Discussion by rpmcnally** [python-cmd2/cmd2#1226] [Q&A]
We are starting development on a new CLI using cmd2 and really like all the features it has to offer.  One thing we have noticed is that long help strings don't wrap very well; our devs have different terminal width preferences so hardcoded line breaks aren't a great solution.  It seems like this can largely be fixed by having the Cmd2HelpFormatter inherit from argparse.HelpFormatter instead of argparse.RawTextHelpFormatter, but I was curious if that would break other things.  I haven't noticed any issues in my testing but the 'Raw' version was probably chosen for a reason.

Also, that doesn't fix the top-level help, which seems to always target 100 columns (though actually needs 102 because of the spaces between columns).  Am I missing some easy way to configure this or have it adjust based on the terminal width?

**Comment by tleonhardt** [comment]
@kmvanbrunt Do you have any thoughts on this?

**Comment by anselor** [comment]
As I recall, the reason we used `RawTextHelpFormatter` was so it would preserve line breaks the developer put in the help text. The standard `HelpFormatter` would strip all line breaks and re-wrap as one continuous text block.

**Comment by anselor** [comment]
I think a general solution for this situation is we should define a constant for the default helpformatter used in cmd2. A developer can then overwrite this constant early in program startup with one of their own.

As a near-term work-around you can probably assign over the Cmd2HelpFormatter with a custom one you've provided. 

something like this *might* do it.
```
import argparse
import cmd2.argparse_custom

OrigCmd2HelpFormatter = cmd2.argparse_custom.Cmd2HelpFormatter

class YourCustomHelpFormatter(argparse.HelpFormatter):
    ... [your custom stuff here]

cmd2.argparse_custom.Cmd2HelpFormatter = YourCustomHelpFormatter

```
